src/common/HeartbeatMap.cc

```diff
@@ -121,6 +121,9 @@ bool HeartbeatMap::is_healthy()
   if (now < m_inject_unhealthy_until) {
     ldout(m_cct, 0) << "is_healthy = false, injected failure for next " << (m_inject_unhealthy_until - now) << " seconds" << dendl;
     healthy = false;
+    m_rwlock.put_read();
+    ldout(m_cct, 20) << "is_healthy = NOT HEALTHY" << dendl;
+    return healthy;
   }
```

Comment:
what about just adding a **} else {** here so that the loop is run only if **now >= m_inject_unhealthy_until** ? It's less change and does the same. Maybe easier to read as well ?